internal/pkg/earpointer/state_test.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package earpointer_test
+
+import (
+	"go/ast"
+	"go/importer"
+	"go/parser"
+	"go/token"
+	"go/types"
+	"sort"
+	"testing"
+
+	"github.com/google/go-flow-levee/internal/pkg/earpointer"
+	"golang.org/x/tools/go/ssa"
+	"golang.org/x/tools/go/ssa/ssautil"
+)
+
+// parseSourceCode parses the source code, convert it to SSA form, and return the SSA package.
+// The filename for source is always "test.go", and the package is "t" at an empty path.
+func parseSourceCode(src string) (*ssa.Package, error) {
+	// Parse the source files.
+	fset := token.NewFileSet()
+	f, err := parser.ParseFile(fset, "test.go", src, parser.ParseComments)
+	if err != nil {
+		return nil, err
+	}
+	files := []*ast.File{f}
+
+	// Create the type-checker's package.
+	pkg := types.NewPackage("t", "")
+	// Type-check the package, load dependencies.
+	// Create and build the SSA program.
+	ssaPkg, _, err := ssautil.BuildPackage(
+		&types.Config{Importer: importer.Default()}, fset, pkg, files, ssa.SanityCheckFunctions)
+	if err != nil {
+		return nil, err
+	}
+	return ssaPkg, nil
+}
+
+// Test basic state operations using single global variables.
+// No fields are involved.
+func TestBasic(t *testing.T) {
+	code := `package p;
+	type T struct {}
+	var g1 *int
+	var g2 *int
+	var g3 T
+	`
+	pkg, err := parseSourceCode(code)
+	if err != nil {
+		t.Fatalf("compilation failed: %s", code)
+	}
+	state := earpointer.NewState()
```

Comment:
These tests are very verbose. Is there a way to shorten them and make them for intuitive?

Reply:
Simplify a little bit.